modules/iam-role-for-serviceaccount/main.tf

```diff
@@ -26,7 +26,9 @@ resource "aws_iam_role" "irsa" {
 }
 
 resource "aws_iam_role_policy_attachment" "irsa" {
-  for_each   = toset(var.enabled ? var.policy_arns : [])
+  for_each = {
+    for key, val in var.policy_arns : key => val
+  }
```

Comment:
refer to https://discuss.hashicorp.com/t/for-each-value-depends-on-resource-attributes-that-cannot-be-determined-until-apply/6061